addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -35,24 +39,207 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
+            
+        }
+    },
+
+    /**
+     * @private
+     */
+    _onClickRemoveLocation: async function (ev) {
+        ev.stopPropagation();
+        const self = this;
+        await this._rpc({
+            route: "/shop/access_point/set",
+            params: {
+                access_point_encoded: null,
+            },
+        })
+        await self._getCurrentLocation();
+        this._disablePayButtonNoPickupPoint(ev);
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onClickShowLocations: async function (ev) {
+        const self = this;
+        const show_pickup_locations = ev.currentTarget;
+        let modal = show_pickup_locations.nextElementSibling;
+        if (!show_pickup_locations.getElementsByClassName("link-primary").length)
+            return;
+        const should_load_content = modal.firstChild ? false : true;
+        while (modal.firstChild) {
+            modal.lastChild.remove();
+        }
+        if (!should_load_content)
+            return;
 
+        var delivery_type = ev.currentTarget.closest(".o_delivery_carrier_select").childNodes[1].attributes.getNamedItem("delivery_type").nodeValue;
+        
+        const delivery_type_id = ev.currentTarget.closest(".o_delivery_carrier_select").childNodes[1].value;
+        await this._checkCarrier(ev,delivery_type_id)
+        
+        $(qweb.render(delivery_type + "_pickup_location_loading")).appendTo($(modal));
+        const data = await self._rpc({
+            route: "/shop/access_point/close_locations",
+        })
+        if (modal.firstChild)
+            modal.firstChild.remove();
+        if(!data || data && data.error || (data && data.close_locations && data.close_locations.length==0)){
+            const error_message = document.createElement("em");
+            error_message.classList.add("text-error");
+            error_message.innerText = data.error ? data.error : "No available Pick-Up Locations";
+            modal.appendChild(error_message);
+            return;
+        }
+        
+        var list_to_render = delivery_type + "_pickup_location_list";
+        var data_to_render = {partner_address: data.partner_address};
+        data_to_render[delivery_type + "_pickup_locations"] = data.close_locations;
+        $(qweb.render(list_to_render, data_to_render)).appendTo($(modal));
+        this._displayCarrierDropper(ev);  
+    },
+    /**
+     * @private
+     * @param {Object} obj // can be either an event or an objet returned by the controller
+     */
+    _displayCarrierDropper: async function (obj) {
+        let current_id = obj.carrier_id
+        const show_locations = document
+            .getElementById("delivery_carrier")
+            .getElementsByClassName("o_show_pickup_locations");
+
+        // We check if it's an event
+        if('currentTarget' in obj){
+            const is_full_page_event = !obj.currentTarget.closest(".o_delivery_carrier_select")
+            if(is_full_page_event)return;
```

Comment:
We try to avoid one line returns
```suggestion
            if (is_full_page_event) {
                return;
            }
```